fondant/component.py

```diff
@@ -223,17 +224,70 @@ def transform(self, *args, **kwargs) -> dd.DataFrame:
             kwargs: Arguments provided to the component are passed as keyword arguments
         """
 
+    @abstractmethod
     def _process_dataset(self, manifest: Manifest) -> dd.DataFrame:
         """
-        Creates a DataLoader using the provided manifest and loads the input dataframe using the
-        `load_dataframe` instance, and  applies data transformations to it using the `transform`
-        method implemented by the derived class. Returns a single dataframe.
+        Load the data based on the manifest using a DaskDataloader and call the transform method to
+        process it.
 
         Returns:
             A `dd.DataFrame` instance with updated data based on the applied data transformations.
         """
         data_loader = DaskDataLoader(manifest=manifest, component_spec=self.spec)
         df = data_loader.load_dataframe()
         df = self.transform(dataframe=df, **self.user_arguments)
+        return df
+
+
+class PandasTransformComponent(TransformComponent):
+    @abstractmethod
+    def setup(self, *args, **kwargs):
```

Comment:
Nice :) should this be moved over to the general `Component` class

Reply:
No, this is only needed in the `PandasTransformComponent` since its `transform` method is called multiple times. The `TransformComponent`'s `transform` method is only called once.

Reply:
ok thanks for clarifying